File: workspace/2023/2023-02-27/fizz-buzz.ruby/fizz_buzz.rb

def main
    puts fizz_buzz(100)
end

def fizz_buzz(n)
    (1..n).map do |i|
        fizz_buzz_value(i)
    end
end

def fizz_buzz_value(i)
    case
        when i % 15 == 0
            'FizzBuzz'
        when i % 5 == 0
            'Buzz'
        when i % 3 == 0
            'Fizz'
        else
            i
    end
end

main

File: workspace/2023/2023-01-18/fizz-buzz.forth/fizz-buzz.fs

: fizz_buzz_value ( n -- )
    dup 15 mod 0 = if
        s" FizzBuzz " type drop
    else
        dup 5 mod 0 = if
            s" Buzz " type drop
        else
            dup 3 mod 0 = if
                s" Fizz " type drop
            else
                .
            endif
        endif
    endif
;

: _fizz_buzz recursive ( max min -- )
    over over < if
        drop drop       ( max min -- )
    else
        dup             ( max min -- max min min )
        fizz_buzz_value ( max min min -- max min )
        1 +             ( max min -- max min+1 )
        _fizz_buzz      ( max min+1 -- )
    endif
;

: fizz_buzz ( n -- )
    1           ( n -- n 1 )
    _fizz_buzz  ( n 1 -- )
;

100 fizz_buzz

